mbedtls fails to build if hal_espressif is missing

### Describe the bug

Something weird is happening since https://github.com/zephyrproject-rtos/zephyr/pull/104031, on some nrf samples I can see a build breakage if the Espressif HAL is not present. To reproduce

```
~/zephyrproject/zephyr$ git show --oneline
5286027c859 (HEAD, origin/main, origin/HEAD, main) drivers: firmware: qemu: don't enable QEMU_FWCFG by default
$ west update
...
$ west build -p -b nrf52dk/nrf52832 samples/bluetooth/peripheral_hr/
...
[372/372] Linking C executable zephyr/zephyr.elf
Memory region         Used Size  Region Size  %age Used
           FLASH:      160868 B       512 KB     30.68%
             RAM:       28620 B        64 KB     43.67%
        IDT_LIST:           0 B        32 KB      0.00%
$ rm -rf ~/zephyrproject/modules/hal/espressif
$ rm -rf build/
$ west build -p -b nrf52dk/nrf52832 samples/bluetooth/peripheral_hr/
...
/home/fabiobaltieri/zephyrproject/modules/crypto/mbedtls/tf-psa-crypto/platform/platform_util.c:240:2: error: #error "No mbedtls_ms_time available"
  240 | #error "No mbedtls_ms_time available"
      |  ^~~~~
/home/fabiobaltieri/zephyrproject/modules/crypto/mbedtls/tf-psa-crypto/platform/platform_util.c:249:2: error: #error "The built-in entropy sources only work on Unix and Windows. " "Please enable MBEDTLS_PSA_DRIVER_GET_ENTROPY instead of " "MBEDTLS_PSA_BUILTIN_GET_ENTROPY and implement " "mbedtls_platform_get_entropy()."
  249 | #error \
      |  ^~~~~
```

### Regression

- [x] This is a regression.

### Steps to reproduce

see above

### Relevant log output

```shell
see above
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

Zephyr SDK 1.0, 5286027c859, Linux

### Additional Context

_No response_

--- a/modules/mbedtls/CMakeLists.txt
+++ b/modules/mbedtls/CMakeLists.txt
@@ -16,6 +16,9 @@ if(CONFIG_MBEDTLS)
     # - It contains public include directories which are provided by Mbed TLS.
     zephyr_interface_library_named(mbedTLS)
 
+    # Explicitly link zephyr_interface to mbedTLS
+    zephyr_link_libraries(mbedTLS)
+
     target_compile_definitions(mbedTLS INTERFACE
       MBEDTLS_CONFIG_FILE="${CONFIG_MBEDTLS_CFG_FILE}"
       TF_PSA_CRYPTO_CONFIG_FILE="${CONFIG_TF_PSA_CRYPTO_CFG_FILE}"

